### DISCUSSION TITLE
Lutron Aurora smart bulb dimmer - odd binding behavior

### DISCUSSION THREAD

**Discussion by GoBooth** [Koenkk/zigbee2mqtt#16248] [Devices]
I have a Lutron Dimmer Z3-1BRL, and I'm experiencing some unexpected behavior.

This dimmer is in my kitchen, and I have the following zigbee devices:

- 3x Hue:  white filament Edison ST72 E27 LED warm-to-cool
- 5x Acuity Brands Lighting: WF4C_WF6C 4" Potlights
- 1x Gledopto CW WW LED Controller

The ultimate problem i'm trying to resolve is:  my dimmer is controlling all lights **except** for 2 of the 5 potlights.

This Dimmer is the first device I ever Binded using Z2M.

At first, I created Groups in Z2M, 1 Group for the 3 Hue Bulbs, and then 1 group for the 5 pot lights.
When I tried Binding the Dimmer to the hue Group, I was NOT able to control the lights.
instead, if I added each bulb individually, I was then able to turn the lights on/off and dim up/down.
I then added all 5 spotlights individually
and then added the single led controller.

This worked OK, except that 2 of the 5 potlights don't respond (they don't respond from the GUI either... or at least, not right away)

anyways, I have UNBINDED all lights/groups to the dimmer except for the 1x Gledopto controller.
I have restarted HA and all the jazz... what i'm seeing is that pressing ON/OFF and even dimming the lights from the Lutron dimmer still works on the Hue lights, the pot lights and the gledopto controller.
if I unbind the controller so that the dimmer has nothing binded to it. ON/OFF Dimming does nothing (as expected)
if I rebind just a single hue Bulb.  the Dimmer controlls ALL bulbs + 3/5 potlights + controller.

Why is that?

does it have something to do with groups?

I notice something similar in my bed room too.

I created a Group in Z2M with my Wife and I's lamp's Hue Bulb + a gledopto controller for the bed's headrest.
I binded a xiaomi button to HER bulb only. but a single press turns on both my light and hers + the Gledopto. is this how Groups are supposed to work?

When Binding, I'm only adding "LevelCtrl" & "OnOff"

I just tested deleting all my Groups in Z2M, and the dimmer with just the 1 bulb BINDED to it, pressing ON/OFF **still** turns on/off everything in the kitchen, but the dimmer now only dims that 1 bulb.

WTF.

**Comment by GoBooth** [comment]
no one?

**Comment by hogdanish** [comment]
Can confirm this is still an issue a year later. I managed to get everything working, albeit with some caveats. For anyone looking into how to get a Aurora smart bulb dimmer working with zigbee2mqtt (on HomeAssistant or otherwise):

- You can put it into pairing mode by quickly tapping the button two times, removing the battery, and tapping the button two times. 
- If this doesn't work, you can factory reset it by quickly tapping the button three times and holding for around 20 seconds on the third tap. Then, when the light on the Aurora begins rapidly blinking, press it again a bunch of times again. If your Lutron Aurora locks up/stops working during the binding process, this is a quick way to reset it and fix everything.
- **You can't bind the Lutron Aurora to groups**. No idea why, but it still doesn't seem possible. Once the Lutron Aurora is connected via zigbee2mqtt, only bind the Lutron Aurora to the LevelCtr cluster of the individual bulbs you want to control with it. 
- In my experience, accidentally binding the LutronAurora to anything besides LevelCtr locks up the dimmer and can only be fixed with the aforementioned factory reset process outlined above.

**Reply by kosmonot** [reply]
Thanks for these invaluable tips! Not having these dimmers functional was a show stopper for me and I was reluctantly considering migrating back to ZHA. Binding the lights individually worked a treat. So nice to have all my devices finally functioning well on z2m ember.

**Reply by tabula-raza** [reply]
Still an issue (binding the Lutron aurora to a group in Z2M reports as successful but doesn't actually work if you push the button/rotate the dial).

The tips still work. Factory reset the aurora, the bulbs, re-add to Z2M, and then bind each bulb to only the referenced cluster. 

Aurora works without zigbee controller online (with minimal lag between button actuation and light responding), and states are updated in Z2M and HASS with only a slight lag.

**Comment by califlower** [comment]
Some other things I've noticed, binding ONLY works if the lutron aurora is "on", as in it will keep retrying until you've pressed the button otherwise it will error after trying to set bindings after about 1 minute

Groups half work, as in the sort of a little bit control the lights but not really. It's completely broken so the feature basically doesn't work. Binding to on/off does work for me and binding to levelctrl also seems to work fine. Overall the experience is pretty poor and mostly undocumented which seems insane for such a popular little switch

**Reply by h14h** [reply]
Wow thank you so much for this. I was driving myself mad trying to understand why I couldn't bind the Aurora to my lights -- never would have occurred to me I needed to walk over and nudge it to turn it "on" to get things to take.

**Comment by antonrohr** [comment]
Even after a factory reset, the dimmer keeps controlling one device that I don't want it to.

**Reply by kosmonot** [reply]
I had this problem too with my Hue lights and the Aurora dimmer (binding directly to individual lights incorrectly).  What's happening seems to be that the lights' own zigbee network remembers the dimmer and pairs to it first.   Any old bindings you had seem to be remembered and perhaps may even be corrupted a bit.  The solution for me was to cut the power to all the lights so the network was inactive and then factory reset the dimmer and pair directly to the zigbee coordinator.  Then turn the power back on for the lights and bind as usual.  Kind of a pain but it worked for me. Twice. YMMMV

**Reply by antonrohr** [reply]
Highly appreciate the response! 
One of the devices that won’t forget about the Lutron Aurora is an under wall dimmer module (Iluminize). It is on the same circuit breaker as the raspberry pi with the zigbee controller. So it’s quite a hassle to turn them off independently. 

I still have a hue bridge, so I’ll try connecting the aurora to it and then back to z2m. Or maybe connect the dimmer to the hue bridge and back. I’ll update with my findings

**Reply by antonrohr** [reply]
Adding the Aurora back to Hue did not help. I really think the individual Hue lights and the Iluminize dimmer remember the Aurora switch. 
Because what did work: Removing the lights from z2m and then adding them again. (No connection to Hue bridge necessary)

**Reply by kosmonot** [reply]
Great to hear you solved it.  Those corrupted bindings are infuriating. The lights definitely conspire together to remember devices that were once bound to them. But it seem almost all zigbee problem are always solvable in the end if you work at them enough from different angles.  I've had good luck using the hue mobile app bluetooth pairing to factory reset lights without the hub.  I forget the exact process, but I think it involved power cycling the light first, then bluetooth pairing with the phone close to the bulb, then resetting with a menu button.
